lib/stm32/l4/rcc.c

```diff
@@ -40,6 +40,8 @@
 #include <libopencm3/stm32/rcc.h>
 
 /* Set the default clock frequencies after reset. */
+/* FIXME: These are never updated.
+ * They depend on RCC_CFGR, RCC_PLLCFGR, RCC_CR's MSIRANGE, etc. */
```

Comment:
drop this please, this is library wide

Reply:
I have not looked around too much, but on one of my other targets, the stm32**f**4, the global variables are updated according to the pll and mux settings of the library calls, cf. https://github.com/libopencm3/libopencm3/blob/e7cf9362efe7d7cb6d225cede08252b990df558a/lib/stm32/f4/rcc.c#L851

I am a bit puzzled. I can of course remove the fixme from this patch but can you please elaborate on the frequency handling: Is the stm32f4 code wrong? Does this fixme apply to most/many/almost all of the other targets? Is this tracked anywhere and is there an established way to add frequency getters?

Reply:
If there's a clock setter routine, it should update those variables, if not, that's an independent bug, and not something I want comments scattered around for.

Reply:
OK, so I'll remove the fixme and report it as extra bug on github(?)

Reply:
only if there _is_ a clock setter.  At present, for L4, there _is not_ so this is _not a bug_ (yet)  L4 doesn't (yet) have a routine like: https://github.com/libopencm3/libopencm3/blob/master/lib/stm32/f4/rcc.c#L789 that takes one of the structs.
_those_ routines update those global vars.

Reply:
(so just drop the comment as requested, it's irrelevant.  You can file a new PR that adds the clock helpers.... but we don't need a bug saying there's no clock helpers...)

Reply:
Oh, I see, so the PLL setters are not supposed to fix these variables at all - only the setup functions. That's quite surprising and had cost me some time to figure out why my UART output is broken ;)
It would be great if there would be some place to document all of this stuff (for devs and users, respectively) - is there?

Reply:
well, for ~everyone else, all the methods come in at the same time, so there's never any users using the raw helpers.  I have no idea why L4 ended up half way.  Should _allllll_ the helpers update those variables? perhaps, but that's that not what is done in the rest of the library right now at least.